plugin/connect.go

```diff
@@ -19,18 +18,28 @@ import (
 //    var fooFunc func()
 //    plugin.Connect("sdk", "Foo", &func)
 func Connect[T any](slug string, name string, target *T) {
-	path, err := pluginPath(slug)
+	// the two locations plugins can be found in are the current working
+	// directory and the nextmv library path
+	paths, err := potentialPluginPaths(slug)
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "err getting plugin path: %v", err)
+		fmt.Fprintf(os.Stderr, "err getting plugin paths: %v", err)
 	}
-	if _, err := os.Stat(path); errors.Is(err, os.ErrNotExist) {
-		fmt.Fprintf(os.Stderr, "plugin file %q does not exist\n\n", path)
+	pluginPath := ""
+	for _, path := range paths {
+		if _, err = os.Stat(path); !errors.Is(err, os.ErrNotExist) {
+			pluginPath = path
+			break
+		}
+	}
+	if pluginPath == "" {
+		fmt.Fprintf(os.Stderr,
+			"could not find plugin %q in any of the paths %q\n\n", slug, paths)
 		os.Exit(1)
 	}
 
-	p, err := loadPlugin(slug, path)
+	p, err := plugin.Open(pluginPath)
```

Comment:
So we essentially attempt to load the first file we find / that exists, right? I like it. :+1: